packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -28,33 +29,44 @@ import { getModifierKeyArray } from './utils';
 
 interface Props extends CommonInteractionProps {
   id?: string;
-  transformSelection?: (selection: Selection) => Selection;
+  transform?: (rawSelection: Selection) => Selection;
+  validate?: (selection: Selection) => boolean;
   onSelectionStart?: () => void;
   onSelectionChange?: (
     selection: Selection | undefined,
-    rawSelection: Selection
+    rawSelection: Selection,
+    isValid: boolean
   ) => void;
-  onSelectionEnd?: (selection: Selection) => void;
-  children: (selection: Selection, rawSelection: Selection) => ReactNode;
+  onSelectionEnd?: (selection: Selection, isValid: boolean) => void;
+  onValidSelection?: (selection: Selection) => void;
+  children: (
+    selection: Selection,
+    rawSelection: Selection,
+    isValid: boolean
+  ) => ReactNode;
 }
 
 function SelectionTool(props: Props) {
   const {
     id = 'Selection',
     modifierKey,
     disabled,
-    transformSelection = (selection) => selection,
+    transform = (selection) => selection,
+    validate = ({ html }) => Box.fromPoints(...html).hasMinSize(20),
```

Comment:
Yeah. Given the general character of `SelectionTool`, I am not sure it makes sense to have a default validation. It may be weird for first-time consumers of the component.

I would put this validation behaviour in the components where it is needed.

Reply:
Interesting thought, you're probably right. I'll move it out :+1: